framework/console/controllers/MigrateController.php

```diff
@@ -196,17 +196,31 @@ protected function createMigration($class)
      */
     protected function getMigrationHistory($limit)
     {
-        if ($this->db->schema->getTableSchema($this->migrationTable, true) === null) {
+       if ($this->db->schema->getTableSchema($this->migrationTable, true) === null) {
             $this->createMigrationHistoryTable();
         }
         $query = new Query;
         $rows = $query->select(['version', 'apply_time'])
             ->from($this->migrationTable)
-            ->orderBy('apply_time DESC, version DESC')
             ->limit($limit)
             ->createCommand($this->db)
             ->queryAll();
-        $history = ArrayHelper::map($rows, 'version', 'apply_time');
+        $history = [];
+        foreach ($rows as $row) {
+            if (preg_match('/m?(\d{6}_?\d{6})(\D.*)?$/is', $row['version'], $matches)) {
+                $time = str_replace('_', '', $matches[1]);
+                $history['m' . $time . $matches[2]] = $row;
+            }
+        }
+        // sort history in desc order
+        uksort($history, function ($a, $b) {
+            if ($a == $b) {
+                return 0;
+            } else {
+                return ($a > $b) ? -1 : 1;
+            }
+        });
+        $history = ArrayHelper::map($history, 'version', 'apply_time');
         unset($history[self::BASE_MIGRATION]);
```

Comment:
`$history[self::BASE_MIGRATION]` is no longer a valid key. See [this](https://github.com/yiisoft/yii2/pull/12975/files#diff-952bc7045bade5b9e9c44621c5f93c19R210)